CMakeLists.txt

```diff
@@ -3,14 +3,15 @@ cmake_minimum_required(VERSION 3.10 FATAL_ERROR)
 option(MBGL_WITH_CORE_ONLY "Build only the core bits, no platform code" OFF)
 option(MBGL_WITH_COVERAGE "Enable code coverage collection" OFF)
 option(MBGL_WITH_QT "Build Mapbox GL Qt bindings" OFF)
+option(MBGL_QT_STATIC "Build Mapbox GL Qt bindings staticly" OFF)
```

Comment:
Why not make a more general flag like `MBGL_OBJECT_LIBRARY` or something? Looks a bit strange in a group of all `MBGL_WITH_*`.

Reply:
I moved this option from `platform/qt/qt.cmake`. I agree that naming can be better but the general idea is to always build one bundled library. When building a shared library, all mbgl libs can be STATIC. When building a staic library (iOS for example), all mbgl libraries have to be OBJECT because this is the simplest way to generate one final, single file, bundled lib.

Reply:
So, I would leave it like this because the intension is not to buil OBJECT library. The solution is to build temporary OBJECT libraries when the final Qt library is requested to be static. Hence, the naming is quite correct.

Reply:
I do not like polluting global CMake configuration with Qt stuff as there are probably other clients of this, but let's keep it as this for now as the build system anyways needs a bit of cleanup in the future.